eradiate/util/attrs.py

```diff
@@ -460,3 +460,22 @@ def f(instance, attribute, value):
             validator(instance, attribute, value)
 
     return f
+
+def make_validator_nequal_other(other):
+    def validator_nequal_other(instance, attribute, value):
+        """Validates if the validated value is different from the specfied
+        'other' value of the same class instance. Raises an AttributeError
+        if not.
+        This validator should only be used for arguments that can be cast
+        to numpy arrays.
+
+        Parameter ``other`` (attribute of instance)
+            Other attribute, which must be different from 'value'
+
+        """
+        other_value = getattr(instance, other)
+        if np.all(value == other_value):
+            raise AttributeError(f"'{attribute}' must be different "
+                                 f"from '{other}'!")
+
+    return validator_vector_nequal_other
```

Comment:
As we discussed on Slack, I don't think you need to introduce this to implement the validation procedure. I'd rather stick to the decorator-based approach, which doesn't require intermediate components.